ml-agents/mlagents/trainers/torch/networks.py

```diff
@@ -404,12 +407,22 @@ def forward(
         encoded_entity = torch.cat(self_attn_inputs, dim=1)
         encoded_state = self.self_attn(encoded_entity, self_attn_masks)
 
+        flipped_masks = 1 - torch.cat(self_attn_masks, dim=1)
+        num_agents = torch.sum(flipped_masks, dim=1, keepdim=True)
+        if torch.max(num_agents).item() > self._current_max_agents:
```

Comment:
I'm a little concerned with this O(N) operation (`torch.max`) but if it doesn't really affect runtime then it's fine